src/SearchDlg.cpp

```diff
@@ -2423,6 +2443,93 @@ bool CSearchDlg::PreTranslateMessage(MSG* pMsg)
     return false;
 }
 
+LRESULT CSearchDlg::ColorizeMatchResultProc(LPNMLVCUSTOMDRAW lpLVCD)
+{
+    switch (lpLVCD->nmcd.dwDrawStage)
+    {
+        case CDDS_PREPAINT:                      // theme hooks this stage
+            return CDRF_NOTIFYITEMDRAW;
+        case CDDS_ITEMPREPAINT:
+            return CDRF_NOTIFYSUBITEMDRAW;
+        case  CDDS_ITEMPREPAINT | CDDS_SUBITEM:
+            return CDRF_NOTIFYPOSTPAINT | CDRF_NEWFONT;
+        case CDDS_ITEMPOSTPAINT | CDDS_SUBITEM: // use the theme color
+        {
+            if (IsDlgButtonChecked(*this, IDC_RESULTFILES) != BST_CHECKED && lpLVCD->iSubItem == 3)
+            {
+                HDC     hdc             = lpLVCD->nmcd.hdc;
+                RECT    rc              = lpLVCD->nmcd.rc;  // lpLVCD->rcText does not work
+                if (rc.top == 0)

```

Comment:
rc does not work either.
The only way that works reliably is to use LVM_GETSUBITEMRECT to get the rect.
https://learn.microsoft.com/en-us/windows/win32/controls/lvm-getsubitemrect

Reply:
I tested it on Win10 and Win7. It works.
`rc.top == 0` happens when we hover on the sub-item. This doesn't need `AlphaBlend`, as the display is not erased.

Could you tell me when it does not work? Then I can test for that.

Reply:
using Win11, the nmcd.rc when hovering covers not just the subitem but all columns on the right as well. Which then produces the mark rect to be drawn on those columns.

Reply:
I'm not sure if I got you here. On my side, it also has no issue on Win11.
No mater `rc.top` equals 0 or not, it has no issue. See the screen-casts:

1. Win11-hover-no-item-highlight
   ![Win11-hover-no-item-highlight](https://github.com/stefankueng/grepWin/assets/19585474/622fc4e1-4bbb-4e7e-b810-7c4f58a3bdff)

2. Win10-hover
   ![Win10-hover](https://github.com/stefankueng/grepWin/assets/19585474/ee32f724-ad06-43c5-b234-77cb6693abed)

If I missed anything, tell me.

Reply:
![grepwin](https://github.com/stefankueng/grepWin/assets/665591/744f16b9-7292-43ed-a61e-5b062d826d8f)
as you can see, the yellow rect is drawn over the path column.

Reply:
Oh... Yes, and thought that still indicates where to find the match.
Fixed by [7d58678](https://github.com/stefankueng/grepWin/pull/451/commits/7d5867826497c68806ba98aca6a89ee30fb2f994).